StRoot/StFcsWaveformFitMaker/StFcsWaveformFitMaker.cxx

```diff
@@ -459,40 +470,33 @@ int StFcsWaveformFitMaker::Make() {
     if(mEnergySelect[0]==0) return kStOK;  // don't touch energy, directly from MC
 
     //Loop over all hits and run waveform analysis of the choice
-    float res[8];
+    float res[8] = {0};
     TF1* func=0;
     for(int det=0; det<kFcsNDet; det++) {      
 	StSPtrVecFcsHit& hits = mFcsCollection->hits(det);
 	int ehp = det/2;
 	int nhit=hits.size();
 	for(int i=0; i<nhit; i++){ //loop over all hits  	    
-	  
+	    memset(res,0,sizeof(res));
 	  auto start=std::chrono::high_resolution_clock::now();
 
 	  //if we are geting pedestal from data
-	  float ped=0.0;
 	  if(mPedMin>=0){
-	    StFcsHit* hit = hits[i];
-	    int p=0;
-	    int n = hit->nTimeBin();
-	    for(int i=0; i<n; i++){
-	      int tb=hit->timebin(i);
-	      if(tb>=mPedMin) p+=hit->adc(i);
-	      if(tb>=mPedMax) break;
-	    }       
-	    ped = float(p)/(mPedMax-mPedMin+1.0);
-	    mDb->setPedestal(hit->ehp(), hit->ns(), hit->dep(), hit->channel(), ped);
+	    AnaPed( hits[i], res[6], res[7] );
+	    mDb->setPedestal(hits[i]->ehp(), hits[i]->ns(), hits[i]->dep(), hits[i]->channel(), res[6] );
 	    if(GetDebug()>1){
 	      char name[100];
-	      mDb->getName(hit->ehp(), hit->ns(), hit->dep(), hit->channel(),name);
-	      printf("%s Pedestal=%d/(%d-%d+1)=%8.2f\n",name,p,mPedMax,mPedMin,ped);
+	      mDb->getName(hits[i]->ehp(), hits[i]->ns(), hits[i]->dep(), hits[i]->channel(),name);
+	      printf("%s Pedestal (%d-%d+1)=%8.2f\n",name,mPedMax,mPedMin,res[6] );
```

Comment:
In STAR makers you should use LOG_DEBUG instead of printf

Reply:
done